user-interface/src/main/java/life/qbic/datamanager/views/general/contact/AutocompleteContactField.java

```diff
@@ -0,0 +1,178 @@
+
+    nameField = new ComboBox<>();
+    nameField.setAllowCustomValue(true);
+    nameField.addCustomValueSetListener(
+        customValueSet -> customValueSet.getSource().setValue(customValueSet.getDetail()));
+    nameField.setRequired(false);
+    nameField.addClassName("name-field");
+    nameField.setPlaceholder("Please enter a name");
+
+    emailField = new ComboBox<>();
+    emailField.setAllowCustomValue(true);
+    emailField.addCustomValueSetListener(
+        customValueSet -> customValueSet.getSource().setValue(customValueSet.getDetail()));
+    emailField.setRequired(false);
+    emailField.addClassName("email-field");
+    emailField.setPlaceholder("Please enter an email address");
+
+    contactSelection.addValueChangeListener(this::onContactSelectionChanged); //write only
+
+    binder.forField(nameField)
+        .withValidator(it -> !isRequired()
+            || !(isNull(it) || it.isBlank()), "Please provide a name")
+        .withValidator(it -> isNull(it) || !it.isBlank() || emailField.isEmpty(),
+            "Please provide a name") // when an email is provided require a name as well
+        .bind(Contact::getFullName,
+            Contact::setFullName);
+
+    binder.forField(emailField)
+        .withValidator(it -> !isRequired() || !(isNull(it) || it.isBlank()),
+            "Please provide an email address")
+        .withValidator(new EmailValidator(
+            "The email address '{0}' is invalid. Please provide a valid email name@domain.de",
+            !isRequired()))
+        .withValidator(it -> isNull(it) || !it.isBlank() || nameField.isEmpty(),
+            "Please provide an email address") // when a name is provided require an email as well
+        .bind(Contact::getEmail,
+            Contact::setEmail);
+
+    Div layout = new Div(nameField, emailField);
+    layout.addClassName("input-fields");
+    add(contactSelection, layout);
+    setItems(new ArrayList<>());
+    clear();
+  }
+
+  private void updateValidationProperty() {
+    this.getElement().setProperty("invalid", !binder.isValid());
+  }
+
+  private static Div renderContact(Contact contact) {
+    var contactName = new Span(contact.getFullName());
+    contactName.addClassName("contact-name");
+    var contactEmail = new Span(contact.getEmail());
+    contactEmail.addClassName("contact-email");
+    var container = new Div();
+    container.addClassName("contact-item");
+    container.add(contactName, contactEmail);
+    return container;
+  }
+
+  private void onContactSelectionChanged(
+      ComponentValueChangeEvent<ComboBox<Contact>, Contact> valueChanged) {
+    //ignore clearing the combobox or empty selection
+    if (valueChanged.getValue() == null) {
+      return;
+    }
+    if (valueChanged.getValue().isEmpty()) {
+      return;
+    }
+    // update the contact to the selected value
+    setContact(valueChanged.getValue());
+    // clear selection box
+    valueChanged.getHasValue().clear();
```

Comment:
NIT: Nicely done, however i'm confused why the contact combobox does not contain the selected value ❓

Reply:
the combination of name and email shown in the selection box is not necessarily one of the selectable prefills. This would need to be changed based on the individual inputs of name and email. As these fields contain redundant information and the top combobox is only used to select from pre-existing contacts, it does not help and only introduces complexity in the input element.
Upon selecting a pre-defined contact, the details can be edited but the combobox on top does not need to hold a value as the value is shown by the individual fields.